pkg/search/postgres/query/numeric_query.go

```diff
@@ -56,7 +57,7 @@ func parseNumericPrefix(value string) (prefix string, trimmedValue string) {
 func parseNumericStringToFloat(s string) (float64, error) {
 	val, err := strconv.ParseFloat(s, 64)
 	if err != nil {
-		return 0, err
+		return 0, errox.InvalidArgs.Newf("cannot parse %q as a number", s)
```

Comment:
Wait, this is the place where we validate user input? How about doing it on a service level. Not necessary in this PR.

Reply:
Service level sounds like some work on proper error handling. I just closed PRs that tried to support such initiative, as of no interest.